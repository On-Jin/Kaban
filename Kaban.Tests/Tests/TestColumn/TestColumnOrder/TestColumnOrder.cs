using System.Net;
using Kaban.GraphQL.Columns;
using Kaban.Tests.Setup;
using Snapshooter;
using Snapshooter.Xunit;
using Xunit.Abstractions;

namespace Kaban.Tests.Tests.TestColumn.TestColumnOrder;

[Collection(nameof(WebAppCollectionFixture))]
public class TestColumnOrder(WebAppFactory factory, ITestOutputHelper testOutputHelper)
    : TestColumnBase(factory, testOutputHelper)
{
    [Fact]
    public async Task GraphQL_PatchColumn_Index_Negative()
    {
        await ResetAndPopulateForColumnOrder();
        var json = await PatchColumn(HttpClientShadow,
            new PatchColumnInput(1, null, -2),
            HttpStatusCode.InternalServerError);
        PrettyJson(json).MatchSnapshot();
    }

    [Fact]
    public async Task GraphQL_PatchColumn_Index_Over()
    {
        await ResetAndPopulateForColumnOrder();
        var json = await PatchColumn(HttpClientShadow,
            new PatchColumnInput(1, null, 9),
            HttpStatusCode.InternalServerError);
        PrettyJson(json).MatchSnapshot();
    }

    [Fact]
    public async Task GraphQL_PatchColumn_Index_0_to_0()
    {
        await ResetAndPopulateForColumnOrder();
        var json = await PatchColumn(HttpClientShadow,
            new PatchColumnInput(1, null, 0));
        PrettyJson(json).MatchSnapshot();
    }

    [Fact]
    public async Task GraphQL_PatchColumn_Index_0_to_2()
    {
        await ResetAndPopulateForColumnOrder();
        var json = await PatchColumn(HttpClientShadow,
            new PatchColumnInput(1, null, 2));
        PrettyJson(json).MatchSnapshot();
        (await GetQueryBoardsString()).MatchSnapshot(SnapshotNameExtension.Create("FullBoardQuery"));
    }

    [Fact]
    public async Task GraphQL_PatchColumn_Index_1_to_4()
    {
        await ResetAndPopulateForColumnOrder();
        var json = await PatchColumn(HttpClientShadow,
            new PatchColumnInput(2, null, 4));
        PrettyJson(json).MatchSnapshot();
        (await GetQueryBoardsString()).MatchSnapshot(SnapshotNameExtension.Create("FullBoardQuery"));
    }

    [Fact]
    public async Task GraphQL_PatchColumn_Index_8_to_2()
    {
        await ResetAndPopulateForColumnOrder();
        var json = await PatchColumn(HttpClientShadow,
            new PatchColumnInput(8, null, 2));
        PrettyJson(json).MatchSnapshot();
        (await GetQueryBoardsString()).MatchSnapshot(SnapshotNameExtension.Create("FullBoardQuery"));
    }
    
    [Fact]
    public async Task GraphQL_PatchColumn_Index_8_to_0()
    {
        await ResetAndPopulateForColumnOrder();
        var json = await PatchColumn(HttpClientShadow,
            new PatchColumnInput(8, null, 0));
        PrettyJson(json).MatchSnapshot();
        (await GetQueryBoardsString()).MatchSnapshot(SnapshotNameExtension.Create("FullBoardQuery"));
    }
    
    [Fact]
    public async Task GraphQL_PatchColumn_Index_0_to_7()
    {
        await ResetAndPopulateForColumnOrder();
        var json = await PatchColumn(HttpClientShadow,
            new PatchColumnInput(1, null, 7));
        PrettyJson(json).MatchSnapshot();
        (await GetQueryBoardsString()).MatchSnapshot(SnapshotNameExtension.Create("FullBoardQuery"));
    }
}