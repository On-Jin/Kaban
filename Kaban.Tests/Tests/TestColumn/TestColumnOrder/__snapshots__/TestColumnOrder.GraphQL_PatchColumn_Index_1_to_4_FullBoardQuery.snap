{
  "data": {
    "boards": [
      {
        "id": 1,
        "name": "Board Name",
        "columns": [
          {
            "id": 1,
            "name": "Column 0",
            "mainTasks": []
          },
          {
            "id": 3,
            "name": "Column 2",
            "mainTasks": []
          },
          {
            "id": 4,
            "name": "Column 3",
            "mainTasks": []
          },
          {
            "id": 5,
            "name": "Column 4",
            "mainTasks": []
          },
          {
            "id": 2,
            "name": "Column 1",
            "mainTasks": []
          },
          {
            "id": 6,
            "name": "Column 5",
            "mainTasks": []
          },
          {
            "id": 7,
            "name": "Column 6",
            "mainTasks": []
          },
          {
            "id": 8,
            "name": "Column 7",
            "mainTasks": []
          }
        ]
      }
    ]
  }
}
